apps/condo/domains/notification/adapters/firebaseAdapter.js

```diff
@@ -0,0 +1,226 @@
+
+const FIREBASE_CONFIG = conf['FIREBASE_CONFIG_JSON'] && JSON.parse(conf['FIREBASE_CONFIG_JSON'])
+const EMPTY_RESULT = {
+    responses: [],
+    successCount: 0,
+    failureCount: 0,
+}
+const FAKE_ERROR_RESPONSE = {
+    success: false,
+    error: {
+        errorInfo: {
+            code: 'fake-error',
+            message: 'Fake error mesage',
+        },
+    },
+}
+
+const getFakeSuccessResponse = () => ({
+    success: true,
+    messageId: `fake-success-message/${faker.datatype.uuid()}`,
+})
+
+/**
+ * Send push notification to pushToken via app, configured by FIREBASE_CONFIG in .env
+ * Attention! Notifications could only be sent to devices, connected via same PROJECT_ID.
+ * Attempts to send push notifications to devices, connected through different projects will fail.
+ */
+class FirebaseAdapter {
+    constructor (config = FIREBASE_CONFIG) {
+        this.app = null
+
+        if (isEmpty(config)) console.error('Valid FIREBASE_CONFIG_JSON should be provided within .env, and can be retrieved from https://console.firebase.google.com/project/__PROJECT_ID__/settings/serviceaccounts/adminsdk')
+
+        try {
+            this.app = admin.initializeApp({ credential: admin.credential.cert(config) })
+        } catch (error) {
+            console.error('Unable to authorize at FireBase using provided FIREBASE_CONFIG_JSON', error)
+        }
+    }
+
+    prepareNotification ({ title, body } = {}) {
+        if (!title || !body || isEmpty(title) || isEmpty(body)) throw new Error('No notification.title or notification.body')
+
+        return { title, body }
+    }
+
+    prepareNotifications (notificationRaw, data, tokensData = []) {
+        const notification = this.prepareNotification(notificationRaw)
+        const notifications = []
+        const fakeNotifications = []
+
+        tokensData.forEach((pushToken) => {
+            const isFakeToken = pushToken === PUSH_FAKE_TOKEN_SUCCESS || pushToken === PUSH_FAKE_TOKEN_FAIL
+            const target = isFakeToken ? fakeNotifications : notifications
+
+            target.push({
+                token: pushToken,
+                data,
+                notification,
+            })
+        })
+
+        return [notifications, fakeNotifications]
+    }
+
+    injectFakeResults (result, fakeNotifications) {
+        const mixed = { ...result }
+
+        fakeNotifications.forEach(({ token }) => {
+            if (token === PUSH_FAKE_TOKEN_SUCCESS) {
+                mixed.successCount++
+                mixed.responses.push(getFakeSuccessResponse())
+            }
+            if (token === PUSH_FAKE_TOKEN_FAIL) {
+                mixed.failureCount++
+                mixed.responses.push(FAKE_ERROR_RESPONSE)
+            }
+        })
+
+        return mixed
+    }
+
+    async sendNotification ({ notification, tokensData, data } = {}) {
+        console.log('sendNotification:', { notification, tokensData, data })
```

Comment:
don't use console.log! You should use `pino` logger. Look at `sbbol` examples or `apolloErrorFormatter`